lib/Transforms/ForwardStoreToLoad/ForwardStoreToLoad.cpp

```diff
@@ -150,6 +150,82 @@ LogicalResult ForwardSingleStoreToLoad::matchAndRewrite(
   return failure();
 }
 
+bool RemoveUnusedStore::isPostDominated(Operation *potentialOp,
+                                        memref::StoreOp &storeOp) const {
+  if (!dominanceInfo.properlyDominates(storeOp.getOperation(), potentialOp)) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "store of is not properly dominated by potential store\n");
+    return false;
+  }
+
+  // Probably want to relax this at some point in the future.
+  if (storeOp->getBlock() != potentialOp->getBlock()) {
+    LLVM_DEBUG(llvm::dbgs() << "store ops are not in the same block\n");
+    return false;
+  }
```

Comment:
Nit: Maybe add a test case for this.

Reply:
That would be the test `cf_between_stores` in this file: https://github.com/google/heir/pull/959/files#diff-c36b93c852ee5629aff3a5adc1547cc7b9f210f3dc22ce2c4896e00f7afb4607